lib/rubocop/cop/minitest/literal_as_actual_argument.rb

```diff
@@ -18,30 +18,34 @@ module Minitest
       #   assert_equal [1, 2], foo, 'message'
       #
       class LiteralAsActualArgument < Cop
+        include ArgumentRangeHelper
+
         MSG = 'Replace the literal with the first argument.'
 
         def on_send(node)
           return unless node.method?(:assert_equal)
 
           actual = node.arguments[1]
-          range = arguments_range(node)
-          add_offense(node, location: range) if actual.recursive_basic_literal?
+          return unless actual
+
+          add_offense(node, location: all_arguments_range(node)) if actual.recursive_basic_literal?
         end
 
         def autocorrect(node)
           expected, actual, message = *node.arguments
-          arguments = [actual.source, expected.source, message&.source].compact.join(', ')
 
           lambda do |corrector|
+            new_actual_source =
+              if actual.is_a?(RuboCop::AST::HashNode) && actual.source[0] != '{'
```

Comment:
Can you use `hash_type?` and `braces?` methods?

```suggestion
              if actual.hash_type? && !actual.braces?
```

Reply:
Yes! I figured what I had was not idiomatic but I had some trouble finding good docs / examples for parser. Do you have a recommendation?

Will update and squash this in a few minutes.

Reply:
Done

Reply:
Thank you!

>  I figured what I had was not idiomatic but I had some trouble finding good docs / examples for parser

For example, there are the following docs.

- https://docs.rubocop.org/rubocop-ast/index.html
- https://www.rubydoc.info/gems/rubocop-ast/RuboCop/AST
- https://www.rubydoc.info/github/whitequark/parser/Parser

And IMHO, code reading into RuboCop AST's may be useful.